tests/Block/AuditBlockServiceTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Tests\Block;
+
+use SimpleThings\EntityAudit\AuditReader as SimpleThingsAuditReader;
+use SimpleThings\EntityAudit\Revision;
+use Sonata\BlockBundle\Block\BlockContext;
+use Sonata\BlockBundle\Model\Block;
+use Sonata\BlockBundle\Test\AbstractBlockServiceTestCase;
+use Sonata\DoctrineORMAdminBundle\Block\AuditBlockService;
+
+/**
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class AuditBlockServiceTest extends AbstractBlockServiceTestCase
+{
+    private $simpleThingsAuditReader;
+    private $blockService;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->simpleThingsAuditReader = $this->prophesize(SimpleThingsAuditReader::class);
```

Comment:
Can you avoid mocking this and use the real thing instead or is it too cumbersome?

Reply:
Then I would need to mock things inside of `SimpleThingsAuditReader`?

Reply:
I don't know what dependencies it has, that's why I'm asking. If there are too many, just don't do it.

Reply:
```     */
    public function __construct(EntityManagerInterface $em, AuditConfiguration $config, MetadataFactory $factory)
    {
        $this->em = $em;
        $this->config = $config;
        $this->metadataFactory = $factory;
        $this->platform = $this->em->getConnection()->getDatabasePlatform();
        $this->quoteStrategy = $this->em->getConfiguration()->getQuoteStrategy();
    }
```

Too many :sweat_smile: , wouldn't that basically be testing 3rd party code?

Reply:
Yeah drop it for that particular case, it's way too complicated. The goal is to avoid mocking internal calls, catch `trigger_error` and deprecations if any